tower-buffer/tests/buffer.rs

```diff
@@ -83,29 +84,55 @@ fn when_inner_is_not_ready() {
 
 #[test]
 fn when_inner_fails() {
+    use std::error::Error as StdError;
+
     let (mut service, mut handle) = new_service();
 
     // Make the service NotReady
     handle.allow(0);
-    handle.error("foobar");
+    handle.error(Error("foobar"));
 
     let mut res1 = service.call("hello");
 
     // Allow the Buffer's executor to do work
     ::std::thread::sleep(::std::time::Duration::from_millis(100));
     with_task(|| {
         let e = res1.poll().unwrap_err();
-        if let error::Error::Closed(e) = e {
-            assert!(format!("{:?}", e).contains("poll_ready"));
-            assert_eq!(e.error(), &tower_mock::Error::Other("foobar"));
+        if let Some(e) = e.downcast_ref::<error::ServiceError>() {
```

Comment:
Most of this junk will go away once `tower-mock` is updated to use `Box<Error>`.